linearregression/Lib.idr

```diff
@@ -0,0 +1,51 @@
+module Lib
+import Types
+
+
+public export
+calculateDeltas : (Double,Double) -> (Double, Double) -> Double
+calculateDeltas (t0, t1) (x, y) = t0 + t1 * x-y
+
+
+
+avg : List Double -> Double
+avg [] = 0.0
+avg xs = do
+  let x = length xs
+  let y = natToInteger x
+  let z = fromInteger y
+  (sum xs) / z
+
+
+
+getmult: List Double -> List Double -> List Double
+getmult [] [] = []
+getmult (x::xs) [] = (x::getmult xs [])
+getmult [] (y::ys) = (y::getmult [] ys)
+getmult (x::xs) (y::ys) = (x * y :: getmult xs ys)
+
+public export
+adjustDeltas : List Double -> List (Double,Double) -> List Double
+adjustDeltas deltas examples =
+  let xs = (map (\(x,y) => x) examples)
```

Comment:
I believe you can simplify the following with

```suggestion
  let xs = map fst examples
```

Reply:
Good point